src/ng/http.js

```diff
@@ -71,14 +71,16 @@ function headersGetter(headers) {
   var headersObj = isObject(headers) ? headers : undefined;
 
   return function(name) {
+    var existingHeaderName, lowercaseName = lowercase(name);
```

Comment:
Can you please move this directly above the `for .. in`? Or even inside, browsers should be able to optimize this.